src/shared/sleep-config.c

```diff
@@ -280,22 +284,49 @@ static int read_resume_files(char **ret_resume, uint64_t *ret_resume_offset) {
         return 0;
 }
 
-static bool location_is_resume_device(const HibernateLocation *location, const char *sys_resume, const uint64_t sys_offset) {
+/*
+ * Determine if the HibernateLocation matches the resume= (device) and resume_offset= (file).
+ *
+ * Returns:
+ *  1 - "yes"   HibernationLocation resume device & resume_offset are equal to sys_resume & sys_offset.
+ *  0 - "no"    HibernateLocation resume device & resume_offset are not equal to sys_resume & sys_offset.
+ * -1 - "maybe" HibernateLocation resume_offset is 0 and sys_offset is > 0 and sys_resume is set.
+ */
+static int location_is_resume_device(const HibernateLocation *location, const char *sys_resume, const uint64_t sys_offset) {
         assert(location);
         assert(location->resume);
         assert(sys_resume);
 
-        return streq(sys_resume, location->resume) && sys_offset == location->resume_offset;
+        bool resume_is_set = !streq(sys_resume, "0:0");
```

Comment:
Hmm, can't we change thing so that instead of string comparing sys_resume with "0:0" everywhere we just say that sys_resume set to NULL means what "0:0" currently means? i.e. in read_resume_files() already filter our "0:0" and set *ret_resume to NULL when we see it.

Propagating this magic "I am not set" string through all layers is just weird.

Reply:
Actually thinking about this, read_resume_files() should probably change ret_resume to be of type dev_t, and use parse_dev() to parse it to dev_t directly? i.e. we should parse where we read the data, and then prpoagate parsed data, instead of literal strings